G/GAP/bundled/patches/gcc12.patch

```diff
@@ -0,0 +1,23 @@
+commit 6f851174fe92c077e316005612e7246819434e0e
+Author: Max Horn <max@quendi.de>
+Date:   2022-09-14 00:12:02 +0200
+
+    kernel: workaround GCC 12 warning
```

Comment:
Wait, are we building with GCC 12?

Reply:
Not saying that, but the only place I've seen this warning before and can reproduce it is with GCC 12, not with GCC 11, so I cherry-picked that patch for use here.

It appeared only in the builds for macos-aarch64, see e.g. [this log](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=22159&view=logs&j=5d923a24-afbc-5e65-3a9d-e5b7b46b446b&t=7772c3a4-c19d-58f1-8481-c85c5a6fe3fe) and there only when invoking `/opt/bin/x86_64-linux-musl-cxx11/x86_64-linux-musl-gcc` (i.e. to compile code to be run during build time)

Reply:
This is the native / build compiler (targeting linux), isn't that always gcc? Anyway I am not aware of anything in that recipe changing the compiler